timestamp-impl/src/main/java/com/palantir/timestamp/TimestampAllocationFailures.java

```diff
@@ -32,6 +32,7 @@
     private final Logger log;
 
     private volatile Throwable previousAllocationFailure;
```

Comment:
Probably doesn't need to be volatile now that you've added EMRT.

responseTo - lock, read PAF, write PAF, unlock, write EMRT
verify - read EMRT, read PAF

So if we've written true to EMRT (in responseTo), we'll have flushed some PAF in to verify already.